packages/framework/collabspaces/src/collabSpaces.ts

```diff
@@ -636,42 +723,35 @@ export class CollabSpacesRuntime
 		return this.createCollabChannel(value, channelId);
 	}
 
-	private mapChannelToCellCore(channelId: string) {
+	private parseChannelId(channelId: string): { rowId: string; colId: string; iteration: string } {
 		const parts = channelId.split(",");
-		assert(parts.length === 3, "wrong channel ID");
+		assert(parts.length === 3, "Invalid channel Id");
 		const rowId = parts[0];
 		const colId = parts[1];
 		const iteration = parts[2];
+		return { rowId, colId, iteration };
+	}
 
-		// TBD(Pri0)
-		// This needs to be way more efficient by building a reverse mapping and maintaining it through all operations!
-		// Need to reconcile which layer does such mapping. At the moment, this is done outside of matrix - in Tablero
-		// component. We should either move it here (and ideally - in compatible way, i.e. not changing document schema),
-		// or move most of the channel logic to the layer that implements row/col stable IDs.
-
-		const rowCount = this.matrix.rowCount;
-		const colCount = this.matrix.colCount;
+	private mapChannelToCellCore(channelId: string) {
+		const { rowId, colId, iteration } = this.parseChannelId(channelId);
+		const row = this.reverseMap.getRowIndex(rowId);
+		const col = this.reverseMap.getColIndex(colId);
 
-		let row;
```

Comment:
Any chance we can have two implementations for remapping - the old logic and new logic? Such that we can easily switch back and forth when investigating issue?

Reply:
If it's too hard, then probably it's Ok not to do it,

Reply:
sure thing. I was planning on leaving the remapping alive and simply do not make use of it when doing the mapChannelToCellCore in case this flag is off. How does it sound ? I could also disable it entirely if the flag is off but I suppose it might be helpful to have it around and debug things in the future.

Reply:
Or do you want to turn it on or off before the test execution ? Like, run this test with the reverseMapping off and then on ? 
Like configure at the factory level ?  I was thinking on a simple global at the collabspace level.